packages/odyssey-react-mui/src/labs/Drawer.tsx

```diff
@@ -0,0 +1,258 @@
+   */
+  showDividers: boolean;
+  /**
+   * The title of the Drawer
+   */
+  title?: string;
+  /**
+   * Type of Drawer
+   */
+  variant?: (typeof variantValues)[number];
+  ariaLabel: string;
+} & HtmlProps;
+
+interface DrawerStyleProps {
+  odysseyDesignTokens: DesignTokens;
+  showDividers: boolean;
+}
+const DrawerHeader = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})<DrawerStyleProps>`
+  position: sticky;
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  top: 0;
+  background-color: ${({ odysseyDesignTokens }) =>
+    odysseyDesignTokens.HueNeutralWhite};
+  margin: 0;
+  padding: ${({ odysseyDesignTokens }) => odysseyDesignTokens.Spacing4}
+    ${({ odysseyDesignTokens }) => odysseyDesignTokens.Spacing5};
+  font-family: ${({ odysseyDesignTokens }) =>
+    odysseyDesignTokens.TypographyFamilyHeading};
+  color: ${({ odysseyDesignTokens }) => odysseyDesignTokens.HueNeutral900};
+  border-bottom: ${({ showDividers, odysseyDesignTokens }) =>
+    showDividers ? `1px solid ${odysseyDesignTokens.HueNeutral200}` : "none"};
+`;
+
+const DrawerContentWrapper = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})<{
+  odysseyDesignTokens: DesignTokens;
+}>`
+  overflow-y: auto;
+`;
+
+const DrawerContent = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})<DrawerStyleProps>`
+  padding: ${({ showDividers, odysseyDesignTokens }) =>
+    showDividers
+      ? `${odysseyDesignTokens.Spacing5}`
+      : `0 ${odysseyDesignTokens.Spacing5}`};
+`;
+
+const DrawerFooter = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})<DrawerStyleProps>`
+  position: sticky;
+  display: flex;
+  justify-content: flex-end;
+  align-items: center;
+  bottom: 0;
+  background-color: ${({ odysseyDesignTokens }) =>
+    odysseyDesignTokens.HueNeutralWhite};
+  padding: ${({ odysseyDesignTokens }) => odysseyDesignTokens.Spacing4};
+  align-content: center;
+  border-top: ${({ showDividers, odysseyDesignTokens }) =>
+    showDividers ? `1px solid ${odysseyDesignTokens.HueNeutral200}` : "none"};
+`;
+
+const Drawer = ({
+  ariaLabel,
+  children,
+  isOpen,
+  onClose,
+  primaryCallToActionComponent,
+  secondaryCallToActionComponent,
+  showDividers = false,
+  tertiaryCallToActionComponent,
+  testId,
+  title,
+  translate,
+  variant = "temporary",
+}: DrawerProps) => {
+  const [isContentScrollable, setIsContentScrollable] = useState(false);
+  const drawerContentRef = useRef<HTMLDivElement>(null);
+  const odysseyDesignTokens = useOdysseyDesignTokens();
+
+  //If RTL is set in the theme, align the drawer on the left side of the screen, uses right by default.
+  const { i18n } = useTranslation();
+  const anchorDirection = i18n.dir() === "rtl" ? "left" : "right";
+
+  useEffect(() => {
+    let frameId: number;
+
+    const handleContentScroll = () => {
```

Comment:
I believe I asked about this previously in your PR but I can't seem to find the answer to it.. what are the performance implications of using this? Would an [intersection observer](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API) be more performant?

Reply:
@chrispulsinelli-okta My understanding is this logic will only trigger if a user us actively scrolling within the drawer or resizing the viewport, which in most cases would be rare and situational (reading text, scrolling to the bottom of a filtering form).

I am open to ideas on updating. I am not familiar with intersection observers but if you have thoughts I'm happy to chat! This logic is mostly pulled from our current implementation of `Dialog` but I added an additional `cancelAnimationFrame(frameId);` to reduce the amount of calculation happening.

Reply:
Here's an interesting article on it that I feel is worth reading:https://javascript.plainenglish.io/should-i-stop-using-scroll-listeners-aa7b0a5af97c 
To not be a blocker, perhaps let's ship this as-is, but I feel its worth creating a ticket to revisit this anywhere where we're tracking user scrolling behaviour for situational UI changes. I can clearly see the advantages to using the intersection observer vs. scroll tracking for performance reasons and while your reasoning above seems fair, I think it's best to adopt a performant pattern now before we run into cases where it's not.

What do you think?